src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -287,54 +284,59 @@ void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
 void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
     OnHostnameResolved(
         absl::StatusOr<std::vector<EventEngine::ResolvedAddress>> addresses) {
+  ValidationErrors::ScopedField field(&errors_, "hostname lookup");
   absl::optional<Resolver::Result> result;
   {
     MutexLock lock(&on_resolved_mu_);
     if (orphaned_) return;
     hostname_handle_.reset();
-    addresses_ = addresses;
+    if (!addresses.ok()) {
+      errors_.AddError(addresses.status().message());
+    } else {
+      addresses_ = std::move(*addresses);
+    }
     result = OnResolvedLocked();
   }
   if (result.has_value()) {
     resolver_->OnRequestComplete(std::move(*result));
   }
-  Unref(DEBUG_LOCATION, "OnHostnameResolved");
 }
 
 void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
     OnSRVResolved(
         absl::StatusOr<std::vector<EventEngine::DNSResolver::SRVRecord>>
             srv_records) {
+  ValidationErrors::ScopedField field(&errors_, "srv lookup");
   absl::optional<Resolver::Result> result;
   auto cleanup = absl::MakeCleanup([&]() {
     if (result.has_value()) {
       resolver_->OnRequestComplete(std::move(*result));
     }
-    Unref(DEBUG_LOCATION, "OnSRVResolved");
   });
-  {
-    MutexLock lock(&on_resolved_mu_);
-    if (orphaned_) return;
-    srv_handle_.reset();
-    if (!srv_records.ok()) {
-      // An error has occurred, finish resolving.
-      balancer_address_lookup_errors_.AddError(srv_records.status().ToString());
-      result = OnResolvedLocked();
-      return;
-    }
-    // Do a subsequent hostname query since SRV records were returned
-    for (const auto& srv_record : *srv_records) {
-      Ref(DEBUG_LOCATION, "OnBalancerHostnamesResolved").release();
-      balancer_hostname_handles_.insert(event_engine_resolver_->LookupHostname(
-          absl::bind_front(
-              &EventEngineDNSRequestWrapper::OnBalancerHostnamesResolved, this,
-              srv_record.host),
-          srv_record.host, std::to_string(srv_record.port),
-          resolver_->query_timeout_ms_));
-      GRPC_EVENT_ENGINE_RESOLVER_TRACE(
-          "DNSResolver::%p Started resolving balancer hostname %s:%d",
-          resolver_.get(), srv_record.host.c_str(), srv_record.port);
-    }
+  MutexLock lock(&on_resolved_mu_);
+  if (orphaned_) return;
+  srv_handle_.reset();
+  if (!srv_records.ok()) {
+    // An error has occurred, finish resolving.
+    errors_.AddError(srv_records.status().message());
+    result = OnResolvedLocked();
+    return;
+  }
+  // Do a subsequent hostname query since SRV records were returned
+  for (const auto& srv_record : *srv_records) {
+    balancer_hostname_handles_.insert(event_engine_resolver_->LookupHostname(
+        [host = srv_record.host,
```

Comment:
Done. We lose the task handle in the logs, but hopefully we won't need them. Since we don't have SourceLocation for any EventEngine callbacks (yet), logs contained TaskHandles have been instrumental in some deep debugging sessions. Down the road, I'd be fine to add LookupTaskHandles to followup logs (or from within the engine implementations we control) if we end up debugging some gnarly problems here.

Reply:
Oh, good point about the task handle being useful in the log.  I think you're right that it would make sense to add a second log message after each lookup is started that includes the task handle.  That way, the logs are easier to understand if the lookup starts in another thread before the log message happens in this thread, but we still get the task handle.

Also, for the log messages that are before we start the lookups, I suggest changing "Started" to "Starting".

Reply:
Both complete.